backend/src/test/java/com/sim_backend/state/StateTest.Java

```diff
@@ -0,0 +1,44 @@
+package com.sim_backend;
+
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+import io.javalin.Javalin;
+import io.javalin.plugin.bundled.CorsPluginConfig;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.*;
+import com.sim_backend.state.*;
+
+public class StateTest {
+
+    private SimulatorStateMachine testStateMachine;
+
+    @BeforeEach
+    void setUp() {
+        testStateMachine = new SimulatorStateMachine();     
+    }
+
+    @Test
+    // Initialize test
+    void testInit() throws Exception{
+        try{
+            assert SimulatorState.PowerOff == testStateMachine.getCurrentState() : "Initialize Failed";
+        } catch (AssertionError e){
+            System.err.println("Catch : " + e.getMessage());
+        }
+    }
+
+    // Booting test Valid Situation
+    void testBooting() throws Exception{
+        try{
+            testStateMachine.transition(SimulatorState.BootingUp);
```

Comment:
For testing state transitions, we want to follow this general flow:

1. Ensure the precondition is true (in this case, check that powered off is the current state)
2. Transition to a state
3. Validate that the postcondition is true (in this case, check that booting up is the current state)
4. Repeat from 1 for all state transitions

This flow is also good testing practice in general.